skyllh/core/utils/multidimgridpdf.py

```diff
@@ -30,6 +31,81 @@ def kde_pdf_bkg_norm_factor_func(pdf, tdm, fitparams):
     """
     return 1. / (2 * np.pi)
 
+def create_MultiDimGridPDF_from_photosplinetable(
+        ds, data, info_key, splinetable_key, norm_factor_func=None,
+        kind=None, tl=None):
+    """Creates a MultiDimGridPDF instance with pdf values taken from KDE PDF
+    values stored in the dataset's auxiliary data.
+
+    Parameters
+    ----------
+    ds : Dataset instance
+        The Dataset instance the PDF applies to.
+    data : DatasetData instance
+        The DatasetData instance that holds the auxiliary data of the data set.
+    info_key : str
+        The auxiliary data name for the PDF information (e.g. axis definitions)
+    splinetable_key : str 
+        The auxiliary data name for the name of the splinetablefile 
+    norm_factor_func : callable | None
+        The normalization factor function. It must have the following call
+        signature:
+            __call__(pdf, tdm, fitparams)
+    kind : str | None
+        The kind of PDF to create. This is either ``'sig'`` for a
+        SignalMultiDimGridPDF or ``'bkg'`` for a BackgroundMultiDimGridPDF
+        instance. If set to None, a MultiDimGridPDF instance is created.
+    tl : TimeLord instance | None
+        The optional TimeLord instance to use for measuring timing information.
+
+    Returns
+    -------
+    pdf : SignalMultiDimGridPDF instance | BackgroundMultiDimGridPDF instance |
+          MultiDimGridPDF instance
+        The created PDF instance. Depending on the ``kind`` argument, this is
+        a SignalMultiDimGridPDF, a BackgroundMultiDimGridPDF, or a
+        MultiDimGridPDF instance.
+    """
+
+
+    if(kind is None):
+        pdf_type = MultiDimGridPDF
+    elif(kind == 'sig'):
+        pdf_type = SignalMultiDimGridPDF
+    elif(kind == 'bkg'):
+        pdf_type = BackgroundMultiDimGridPDF
+    else:
+        raise ValueError('The kind argument must be None, "sig", or "bkg"! '
+            'Currently it is '+str(kind)+'!')
+
+    # Load the PDF data from the auxilary files.
+    num_dict = ds.load_aux_data(info_key, tl=tl)  
+
+    kde_pdf_axis_name_map = ds.load_aux_data('KDE_PDF_axis_name_map', tl=tl)
+    kde_pdf_axis_name_map_inv = dict(
+        [(v,k) for (k,v) in kde_pdf_axis_name_map.items()])
+
+    axis_binnings = [
+        BinningDefinition(
+            kde_pdf_axis_name_map_inv[var], num_dict['bins'][idx])
+        for (idx,var) in enumerate(num_dict['vars'])
+    ]
+ 
+    # get splinetable file
+    splinetable_file_list = ds.get_aux_data_definition(splinetable_key) 
```

Comment:
Why is this a list of files, when there is always only a single file name used?

Reply:
not sure. seems related to the design of the dataset class.
see https://github.com/IceCubeOpenSource/skyllh/blob/master/skyllh/core/dataset.py#L1035
I'll leave this one to you, as it is not related to the splinetable feature.